cypress/integration/common/kiali_login.ts

```diff
@@ -0,0 +1,52 @@
+import { Given, When, And, Then } from "cypress-cucumber-preprocessor/steps";
+
+const USERNAME = 'jenkins';
+const PASSWD = Cypress.env('passwd')
+const KUBEADMIN_IDP = 'my_htpasswd_provider';
+const auth_strategy = JSON.parse(Cypress.env('auth_strategy'))
+
+Given('user opens base url', () => {
+    cy.visit('/'); // visits baseUrl which is set in plugins/index.ts or bdd.ts
+    cy.window().then((win: any) => {
+        if (win = true) {
+            cy.log(win.localStorage + 'skipping login, console is running with auth disabled');
+            return;
+        }
+        // Make sure we clear the cookie in case a previous test failed to logout.
+        cy.clearCookie('openshift-session-token');
+    });
+});
+
+And('user clicks Log In With OpenShift', () => {
+    if (auth_strategy.strategy == 'openshift') {
```

Comment:
Generally doing strict equality with three equals `===` leads to less weirdness. Otherwise javascript will do type conversions for you like `['openshift'] == 'openshift'`.

Reply:
fixed, added strict equals